### ISSUE TITLE
[investigation] Storage lock file complicating deployment with Dokku

### ISSUE THREAD

**Issue by carbontwelve** [superseriousbusiness/gotosocial#596]
I have been working on documenting how to deploy GoToSocial with Dokku and have largely been very successful in doing so. 

The final piece of the puzzle is restarting the service however Dokku implements zero downtime deployments; this means that when restarting the container, a fresh container will be spun up, connectivity checks undertaken and then traffic routed to the fresh container while the old container is scheduled for deletion.

This fails for GoToSocial because the storage lock file mechanism in use, the new container starts up but because it's running at the same time as the old one it sees the existing lock file and bails.

My short term remedy for this is to disable zero downtime deployment for GoToSocial in Dokku, thus meaning Dokku will stop the currently running container before starting a fresh one.

I don't know what to suggest as a long term remedy for this, Zero Downtime deployment is nice to have but I don't mind the few seconds downtime.

Please see this issue as more a seed for discussion than a problem to fix ASAP.

**Comment by igalic** [comment]
from our Config:

```yaml

# String. Type of storage backend to use.
# Examples: ["local", "s3"]
# Default: "local" (storage on local disk)
# NOTE: s3 storage is not yet supported!
storage-backend: "local"
```

okay, so, as soon as we support s3, and dokku offers an s3 endpoint, we're all set!

**Comment by NyaaaWhatsUpDoc** [comment]
I might look for a decent S3 library and write a quick storage.Storage implementation so that we can use it

**Comment by NyaaaWhatsUpDoc** [comment]
Can this ticket be re-addressed now?
